### PR TITLE
Add wavefront obj export

### PR DESCRIPTION
**Pull request by emogenet**
Hi everyone,

Long time OpenSCAD user, and I have lots of command line tools that consume wavefront OBJ file format as input.

Right now, I have to export to STL / OFF / ... and use another tool (eg meshlab) to get an OBJ, which complexifies my workflow. 

So I bit the bullet and added an OBJ exporter. to OpenSCAD.

Please consider adding the feature.

### CODE REVIEW

File: src/gui/MainWindow.cc [hunk 1]
```diff
@@ -2017,6 +2021,8 @@ void MainWindow::sendToOctoPrint()
   const QString fileFormat = QString::fromStdString(Settings::Settings::octoPrintFileFormat.value());
   FileFormat exportFileFormat{FileFormat::STL};
   if (fileFormat == "OFF") {
+    exportFileFormat = FileFormat::OBJ;
+  } else if (fileFormat == "OFF") {
     exportFileFormat = FileFormat::OFF;
```
Comment (t-paul) [inline src/gui/MainWindow.cc:2026]: Typo, one shoudl be `OBJ`.

**Review by t-paul — CHANGES_REQUESTED** [review]
Looks good.

Would you mind adding a simple test case along the lines of this one:
e72b6c412724153a86ca28bd9956fb3636cc0edc

See https://github.com/openscad/openscad/blob/master/doc/testing.txt for some details about the tests. Section "Adding a new test" explains how to generate the reference file automatically.

File: src/gui/MainWindow.cc (hunk 1 above)
Comment (emogenet) [reply src/gui/MainWindow.cc]: typo fixed
tests added
see 4 commits below

**Review by t-paul — APPROVED** [review]
Thanks!

**Pull request merged**